### PR TITLE
[pkg/telemetryquerylanguage] Added Int factory function (#11810)

### PR DESCRIPTION
**Pull request by mcdoker18**
**Description:** 
Added the Int function to convert string and float type to int.
Use cases are described here #11810

**Link to tracking Issue:** #11810

**Testing:** unit tests

**Documentation:** updated the readme file

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
@mcdoker18 Thanks for the contribution. Please split this into 2 PRs, one that adds the function to TQL and then one after that adds the function to the transform processor.

File: pkg/telemetryquerylanguage/functions/tqlcommon/func_int.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"strconv"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
+)
+
+func Int(target tql.Getter) (tql.ExprFunc, error) {
+	return func(ctx tql.TransformContext) interface{} {
+		value := target.Get(ctx)
+		switch value := value.(type) {
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/func_int.go:26]: Can you add support for Boolean?

Comment (mcdoker18) [reply pkg/telemetryquerylanguage/functions/tqlcommon/func_int.go:26]: Done

File: pkg/telemetryquerylanguage/functions/tqlcommon/README.md [hunk 2]
```diff
@@ -27,6 +28,22 @@ Examples:
 
 - `IsMatch("string", ".*ring")`
 
+## Int
+
+`Int(value)`
+
+The `Int` factory function converts a float, bool and string data to the int type.
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/README.md:35]: Can you explicitly list out the types converted, and explicitly state that the returned type is int64

File: pkg/telemetryquerylanguage/functions/tqlcommon/func_int.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			}
+
+			return intValue
+		case float64:
+			return (int64)(value)
+		case bool:
+			if value {
+				return int64(1)
+			}
+
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/func_int.go:42]:
```suggestion
```

File: pkg/telemetryquerylanguage/functions/tqlcommon/func_int_test.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+			value:    "test",
+			expected: nil,
+		},
+		{
+			name:     "int64",
+			value:    int64(333),
+			expected: int64(333),
+		},
+		{
+			name:     "float64",
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/func_int_test.go:53]: Can you add a test for a float that doesn't have a decimal

File: pkg/telemetryquerylanguage/functions/tqlcommon/README.md [hunk 5]
```diff
@@ -27,6 +28,30 @@ Examples:
...
 
+## Int
+
+`Int(value)`
+
+The `Int` factory function converts the `value` to int type.
+
+The returned type is int64.
+
+The input `value` types:
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/README.md:39]: Technically int64 is a supported type. Might want to state somewhere that passing int64 just returns the input with no changes.

File: pkg/telemetryquerylanguage/functions/tqlcommon/func_int_test.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+			value:    float64(2.7),
+			expected: int64(2),
+		},
+		{
+			name:     "float64 without decimal",
+			value:    float64(55.0),
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlcommon/func_int_test.go:59]:
```suggestion
			value:    float64(55),
```

**Review by TylerHelmuth — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by mcdoker18** [comment]
@TylerHelmuth I was planning to add the changelog in the next PR, do you mind?

**Comment by TylerHelmuth** [comment]
> @TylerHelmuth I was planning to add the changelog in the next PR, do you mind?

Please add a changelog entry for the function addition to pkg/telemetryquerylanguage. We'll add a second changelog entry for the transform processor when the processor is updated to register the function. 

We need entries for both because the TQL package can be used independently of the transform processor.

**Review by kentquirk — APPROVED** [review]
One question about the interpretation of strings.

Also, maybe this is out of scope for this PR, but this makes me immediately want equivalent functions for Float() and String() and Bool(). @TylerHelmuth thoughts?

I'm going to approve because I don't think either of these is a blocker.

**Comment by TylerHelmuth** [comment]
@kentquirk ya we probably should have more casting functions, or build casting into the grammar natively.

**Pull request merged**